src/services/ghost/GhostStrategy.ts

```diff
@@ -1,754 +1,84 @@
-	   	* **If the diff shows a variable/function/identifier being renamed** (e.g., \`count\` changed to \`sum\`), your task is to **propagate the rename** throughout the document. Update all references to use the new name. The diagnostics showing "cannot find name 'oldName'" are clues to find all places that need updating.
-	   	* Assume temporary diagnostics (like 'unused variable' or 'missing initializer' on a new line) are signs of work-in-progress.
-	   	* Your task is to **complete the feature**. For an unused variable, find a logical place to use it. For an incomplete statement, finish it. **Never suggest deleting the user's new work or reverting their changes. Always help them move forward.**
-
-	**Rule for DELETIONS:**
-    	* **If the diff shows a line was deleted**, your primary intent is **LOGICAL REMOVAL**.
-    	* Assume the user wants to remove the functionality associated with the deleted code.
-    	* The new diagnostics (like "'variable' is not defined") are not errors to be fixed by re-adding code. They are your guide to find all the **obsolete code that now also needs to be deleted.**
-    	* Your task is to **propagate the deletion**. Remove all usages of the deleted variables, functions, or components.
-
-    * **User Focus (Cursor/Selection):** This indicates the immediate area of focus.
-    
-	* **Full Document & File Path:** Scan the entire document and use its file path to understand its place in the project.
-
-4.  **Strict JSON Array Output Format:** Your entire response **MUST** be a valid JSON array containing objects with the following structure:
-    * Each object must have exactly two properties: "path" and "content"
-    * "path": The full file URI (e.g., "file:///absolute/path/to/file.tsx")
-    * "content": The complete, updated content of that file with proper JSON ESCAPING (quotes as \", newlines as \n, etc.)
-    * **Example:**
-      \`\`\`json
-      [
-        {
-          "path": "file:///Users/catrielmuller/Dev/KiloOrg/example/projects/react/src/App.tsx",
-          "content": "import React from 'react';\\n\\nfunction App() {\\n  return <div>Hello World</div>;\\n}\\n\\nexport default App;"
-        }
-      ]
-      \`\`\`
-    * Do not include any conversational text, explanations, or any text outside of this required JSON format.
-    * Ensure all special characters in the content are properly escaped according to JSON standards.
-    * **Important**: Preserve the exact formatting of the input document, including any empty last lines. If the input document ends with an empty line, your output content must also end with an empty line (represented as \\n at the end).
-`
-
-		return customInstructions ? `${basePrompt}${customInstructions}` : basePrompt
+	private streamingParser: GhostStreamingParser
+	private strategyManager: PromptStrategyManager
+	private debug: boolean
+
+	constructor(debug: boolean = false) {
```

Comment:
#nit lets pass `debug` in as as object so nobody has to guess what `new GhostStrategy(true)` means :)
`new GhostStrategy({ debug: true })`